tst/regression/utils/test_case.py

```diff
@@ -156,34 +148,14 @@ def __checkMPIExecutable(self,mpi_executable):
 
         mpi_exec = mpi_executable[0]
 
-        choices=['mpirun', 'mpiexec', 'srun', 'qsub', 'lsrun', 'aprun']
-        sys.stdout.flush()
-        for choice in choices:
-            sys.stdout.flush()
-            if mpi_exec.endswith(choice):
-                sys.stdout.flush()
-                if len(mpi_exec) != len(choice):
-                    sys.stdout.flush()
-                    if os.path.isfile(mpi_exec):
-                        return
-                    else:
-                        error_msg = "mpi executable path provided, but no file found "
-                        error_msg += mpi_exec 
-                        raise TestManagerError(error_msg)
-                else:
-                    return
-
-        error_msg = "Invalid mpi executable specified "
-        error_msg += mpi_exec 
-        error_msg += "\nExpected choices are:"
-        for choice in choices:
-            error_msg += choice
-            error_msg += "\n"
-        raise TestManagerError(error_msg)
+        if which(mpi_exec) is None:
+            error_msg = "mpi executable path provided, but no file found: "
+            error_msg += mpi_exec
+            raise TestManagerError(error_msg)
```

Comment:
Vast improvement here thanks to moving this logic into the machine specific files and leaving it up to the user.